.github/workflows/ci.yml

```diff
@@ -1,48 +1,52 @@
 name: CI
 
 on: [pull_request]
+env:
+  DEVELOPER_DIR: /Applications/Xcode_13.0.app/Contents/Developer
+  FLUTTER_VERSION: 3.0.2
+  JAVA_VERSION: "12.x"
+  JAVA_DISTRIBUTION: 'zulu'
 
 jobs:
   ci:
-    runs-on: ubuntu-latest
+    runs-on: macos-11
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v2
         with:
-          java-version: "12.x"
-      - uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
+      - uses: subosito/flutter-action@v2
```

Comment:
For 3rd-party actions, we prefer commit hashes instead of version tags, since that makes it less likely for workflow logic to be changed out from under us. It's probably not all that likely that a maintainer or bad actor would do that and actually impact our jobs, but better to be safe than sorry. 🙏